src/shared/TileHOC.jsx

```diff
@@ -0,0 +1,12 @@
+import React, { Component } from 'react';
```

Comment:
I think a better approach would be to stringify classes using mergeClassNames *before* they are passed to Tile component.

Here's why:
* We're getting equality check working out of the box
* We're not expanding component tree even more
* We still compare props other than `classes`, which *is* desirable

Reply:
There are a lot more props that fail the equality check besides just `classes`.  I think fixing all of those issues will be pretty large in scope in comparison to this hack.  I am totally fine if you don't want this hack getting merged in, because the `onMouseLeave`  and `onMouse` performance optimizations solve the majority of the performance bottleneck.  Please let me know and I will just remove this HOC from this pr.